File: Week 6 - Python (P Set 6)/2)Mario/Mario-More.py
# CS50-WEEK6-PSET6-MARIO-MORE
# "Let’s recreate those pyramids (Toward the beginning of World 1-1 in Nintendo’s Super Mario Brothers, Mario must hop over adjacent pyramids of blocks) in C, albeit in text, using hashes (#) for bricks"

import cs50

height = 0

# user input for the height for the pyramid
# while loop checks if the user input is between 1 and 8 (both inclusive)
while height < 1 or height > 8:
    height = cs50.get_int("Height : ")

# prints the pyramid
for i in range(1, height + 1):
    print(" " * (height - i), end="")
    print("#" * i, end="  ")
    print("#" * i)

File: Week 1 - C (P Set 1)/2)Mario/Mario-More.c
//CS50-WEEK1-PSET1-MARIO-MORE
//"Let’s recreate those pyramids (Toward the beginning of World 1-1 in Nintendo’s Super Mario Brothers, Mario must hop over adjacent pyramids of blocks) in C, albeit in text, using hashes (#) for bricks"

#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height;

    //user input for the height for the pyramid
    //do-while loop checks if the user input is between 1 and 8 (both inclusive)
    do
    {
        height = get_int("Height : ");
    }
    while (height < 1 || height > 8);

    //prints the pyramid
    for (int i = 0; i < height; i++)
    {
        //prints the spaces
        for (int j = height - 1; j > i; j--)
        {
            printf(" ");
        }

        //prints the first set of hashes
        for (int j = 0; j <= i; j++)
        {
            printf("#");
        }

        printf("  ");

        //prints the second set of hashes
        for (int j = 0; j <= i; j++)
        {
            printf("#");
        }

        printf("\n");
    }
    return 0;
}
